More detailed setup documents with less assumptions on users (#359)
hound-search/hound: README.md

[Regular Expression Matching with a Trigram Index](http://swtch.com/~rsc/regexp/regexp4.html). Hound itself is a static
[React](http://facebook.github.io/react/) frontend that talks to a [Go](http://golang.org/) backend. The backend keeps an up-to-date index for each repository and answers searches through a minimal API. Here it is in action:

![Hound Screen Capture](imgs/screen_capture.gif)

## Quick Start Guide

### Using Go Tools

> 0. [Install the Go tools](https://golang.org/doc/install) if you don't have it. 

Reviewer: We could probably make this a bit terser:
> 0. [Install Go](https://golang.org/doc/install) if you don't have it already. Hound requires version 1.4 or later. You might also want to define a [`GOPATH` environment variable](https://github.com/golang/go/wiki/GOPATH)) (it defaults to `$HOME/go` if you don't explicitly have one set). If everything is installed properly, `go version` will print out the installed version of go.

Author: Thanks for your suggestion. Updated.

+ After installing Go, make sure go's version is 1.4+ by typing `go version` in the command line. 
+ Set up your [`GOPATH`](https://github.com/golang/go/wiki/GOPATH) to a directory where you want to store all your Go packages. Otherwise
all packages will be installed to `$HOME/go` by default.

1. Use the Go tools to install Hound. The binaries `houndd` (server) and `hound` (cli) will be installed in your $GOPATH.

```
go get github.com/hound-search/hound/cmds/...
```

> 2. Create a [simple-config-example.json](config-examples/simple-config-example.json) in your local file directory. For simplicity, we just showed 

Reviewer: I think that the quick start guide for the most part should be kept relatively brief. The `config-example.json` documents the variety of ways to set up a repo in lieu of more explicit documentation. Since we don't really know what sort of repo people want to track, it's probably a good idea to not make a ton of assumptions. 

Instead, it might be helpful to use the `default-config.json` that's included in the root of the repo:
> Create a `config.json` file and use it to list your repositories. Check out [our example `config.json`](#link) to see how to set up various types of repositories. For example, we can configure Hound to search its own source code using the config found in `default-config.json`:
```json
{
  "dbpath" : "db",
  "repos" : {
    "Hound" : { "url" : "https://github.com/etsy/hound.git" }
  }
}
```

Author: I see. thank you! I did not notice there was a simplified `default-config.json` before.

the simplest configuration for local file search only. (For a more hybrid solution, please refer to [config-example.json](config-examples/config-example.json))

In the simple configuration file, we placed a tensorflow repo at `/home/rfan/tensorflow` path. 

> 3. Run the Hound server with `houndd -addr=localhost:6880 -conf <your config.json file>` and you should see output similar to:

Reviewer: I think I'd also omit the arguments to `houndd`, although I would mention that it needs to be run from the same directory that the config file is in.

Author: Sure. i can remove that.

```
2015/03/13 09:07:42 Searcher started for statsd
2015/03/13 09:07:42 Searcher started for Hound
2015/03/13 09:07:42 All indexes built!
2015/03/13 09:07:42 running server at http://localhost:6080
```

> 4. Open your browser at `http://localhost:6880`, you should be able to search the tensorflow repo rapidly like the following.

Reviewer: To clean things up a bit:
> 4. By default, hound hosts a web ui at `http://localhost:6080 `. Open it in your browser, and start searching.

I don't necessarily think we need another screenshot of the UI here since we include that gif at the top of the README.

Author: Done. Removed.

![Tensorflow Hound Search Screenshot](imgs/tensorflow_hound_search.png)

### Using Docker (1.4+)

0. [Install the docker](https://docs.docker.com/get-docker/) if you don't have it.

> 1. Create a [simple-config-example.json](config-examples/simple-config-example.json) in your local file directory. For simplicity, we just showed 

Reviewer: Same comment here as above.

Author: Updated.

[Regular Expression Matching with a Trigram Index](http://swtch.com/~rsc/regexp/regexp4.html). Hound itself is a static
[React](http://facebook.github.io/react/) frontend that talks to a [Go](http://golang.org/) backend. The backend keeps an up-to-date index for each repository and answers searches through a minimal API. Here it is in action:

![Hound Screen Capture](imgs/screen_capture.gif)

## Quick Start Guide

### Using Go Tools

0. [Install Go](https://golang.org/doc/install) if you don't have it already. Hound requires version 1.4 or later. 
You might also want to define a [`GOPATH`](https://github.com/golang/go/wiki/GOPATH) environment variable) 

> (it defaults to $HOME/go if you don't explicitly have one set). If everything is installed properly, go version will 

Reviewer: nit: 
```suggestion
(it defaults to $HOME/go if you don't explicitly have one set). If everything is installed properly, `go version` should 
```

print out the installed version of go. 

1. Use the Go tools to install Hound. The binaries `houndd` (server) and `hound` (cli) will be installed in your $GOPATH.

```
go get github.com/hound-search/hound/cmds/...
```

2. Create a config.json file and use it to list your repositories. Check out our [example-config.json](config-example.json) 
to see how to set up various types of repositories. For example, we can configure Hound to search its own source code using 
the config found in [default-config.json](default-config.json):
```json
{
  "dbpath" : "db",
  "repos" : {
    "Hound" : { "url" : "https://github.com/etsy/hound.git" }
  }
}
```

> 3. Run the Hound server with `houndd` in the same directory and you should see output similar to:

Reviewer: ```suggestion
3. Run the Hound server with `houndd` in the same directory as your `config.json`. You should see output similar to:
```

```
2015/03/13 09:07:42 Searcher started for statsd
2015/03/13 09:07:42 Searcher started for Hound
2015/03/13 09:07:42 All indexes built!
2015/03/13 09:07:42 running server at http://localhost:6080
```

4. By default, hound hosts a web ui at http://localhost:6080 . Open it in your browser, and start searching.

### Using Docker (1.4+)

0. [Install the docker](https://docs.docker.com/get-docker/) if you don't have it. We need at least `Docker >= 1.14`.

> 2. Create a config.json file and use it to list your repositories. Check out our [example-config.json](config-example.json) 

Reviewer: ```suggestion
1. Create a config.json file and use it to list your repositories. Check out our [example-config.json](config-example.json) 
```

## Running in Production

There are no special flags to run Hound in production. You can use the `--addr=:6880` flag to control the port to which the server binds. 

> Currently, Hound does not support TLS as most users simply run Hound behind either Apache or nginx. Adding TLS support is pretty straight forward though if anyone wants to add it.

Reviewer: Just to rephrase some things while we're here: 
```suggestion
Currently, Hound does not support TLS as most users simply run Hound behind either Apache or nginx. However, we are open to contributions to add TLS support.
```
